src/activities/menu/Menu.qml

```diff
@@ -218,12 +235,24 @@ ActivityBase {
                     }
 
                     function selectCurrentItem() {
-                        particles.burst(10)
-                        ActivityInfoTree.filterByTag(modelData.tag)
-                        ActivityInfoTree.filterLockedActivities()
-                        ActivityInfoTree.filterEnabledActivities()
-                        menuActivity.currentTag = modelData.tag
-                        section.currentIndex = index
+
+
+                               if(modelData.tag === "search"){
```

Comment:
When I change the difficulty, all the activities are displayed by default, that should not be the case, only the activities with the selected difficulties should be displayed.
The code to do this is currently in filterByTag. I would put the common code in a new method filterByDifficulty(int levelMin, int levelMax) and call it from both filterByTag() and filterBySearch().